src/main/java/com/fittracker/fittracker/entity/Product.java

```diff
@@ -0,0 +1,189 @@
+package com.fittracker.fittracker.entity;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
+import jakarta.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+import org.hibernate.annotations.CreationTimestamp;
+
+@Entity
+@Table(name = "products", schema = "fittracker")
+@IdClass(Product.ProductId.class)
+public class Product {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
+    @Id
```

Comment:
What do you think about using:
```
@EmbeddedId
private ProductId productId;
```
while removing `@IdClass(Product.ProductId.class)`?

Reply:
Tried it and had problems setting up generated values. In addition IdClass is usually faster (due to EmbeddedId extra query traversal) https://www.baeldung.com/jpa-composite-primary-keys (Section 5). In addition EmbeddedId is great for natural Indentifiers (Containing set of information that means something on it's own) which is not the case here.

I propose to leave IdClass